contracts/modules/TransferManager/LockUpTransferManager.sol

```diff
@@ -0,0 +1,651 @@
+     * @param _amount Amount of tokens that need to transact
+     */
+    function _checkIfValidTransfer(address _userAddress, uint256 _amount) internal view returns (Result) {
+        uint256 totalRemainingLockedAmount = getLockedTokenToUser(_userAddress);
+        // Present balance of the user
+        uint256 currentBalance = ISecurityToken(securityToken).balanceOf(_userAddress);
+        if ((currentBalance.sub(_amount)) >= totalRemainingLockedAmount) {
+            return Result.NA;
+        }
+        return Result.INVALID;
+    }
+
+    /**
+     * @notice Provide the unlock amount for the given lockup for a particular user
+     */
+    function _getUnlockedAmountForLockup(bytes32 _lockupName) internal view returns (uint256) {
+        /*solium-disable-next-line security/no-block-members*/
+        if (lockups[_lockupName].startTime > now) {
+            return 0;
+        } else if (lockups[_lockupName].startTime.add(lockups[_lockupName].lockUpPeriodSeconds) <= now) {
+            return lockups[_lockupName].lockupAmount;
+        } else {
+            // Calculate the no. of periods for a lockup 
+            uint256 noOfPeriods = (lockups[_lockupName].lockUpPeriodSeconds).div(lockups[_lockupName].releaseFrequencySeconds);
+            // Calculate the transaction time lies in which period
+            /*solium-disable-next-line security/no-block-members*/
+            uint256 elapsedPeriod = (now.sub(lockups[_lockupName].startTime)).div(lockups[_lockupName].releaseFrequencySeconds);
+            // Calculate the allowed unlocked amount per period
+            uint256 amountPerPeriod = (lockups[_lockupName].lockupAmount).div(noOfPeriods);
+            // Find out the unlocked amount for a given lockup
+            uint256 unLockedAmount = elapsedPeriod.mul(amountPerPeriod);
+            return unLockedAmount;
+        } 
+    }
+
+    function _removeLockupType(bytes32 _lockupName) internal {
+        require(lockups[_lockupName].startTime != 0, "Lockup type doesn’t exist");
+        require(lockupToUsers[_lockupName].length == 0, "Users are associated with the lockup");
+        // delete lockup type 
+        delete(lockups[_lockupName]);
+        uint256 i = 0;
+        for (i = 0; i < lockupArray.length; i++) {
+            if (lockupArray[i] == _lockupName) {
+                break;
+            }
+        }
+        if (i != lockupArray.length -1) {
+            lockupArray[i] = lockupArray[lockupArray.length -1];
+        }
+        lockupArray.length--;
+        emit RemoveLockUpType(_lockupName);
+    }
+
+    function _modifyLockUpType(
+        uint256 _lockupAmount,
+        uint256 _startTime,
+        uint256 _lockUpPeriodSeconds,
+        uint256 _releaseFrequencySeconds,
+        bytes32 _lockupName
+    )
+        internal
+    {
+        /*solium-disable-next-line security/no-block-members*/
+        require(lockups[_lockupName].startTime > now, "Not allowed to modify");
```

Comment:
Because a lockup type can be associated with multiple addresses. one change can affect all the address which are associated with a given lockup. that may not be viable during the running of multiple lockups. Which may not be the intention of the issuer to affect all lockups. 

To change a lockup for a particular user, Issuer can remove the previous lockup for him and create a one.
I can create an extra function that will allow modifying the lockup for all holders associated with it with where there is no start time check.

Reply:
`one change can affect all the address which are associated with a given lockup. that may not be viable during the running of multiple lockups. Which may not be the intention of the issuer to affect all lockups.`

It's pretty clear that one lockup can be applicable to multiple people. I don't think it will cause any confusions. If an issuer wants different lockups for different then he should not be using the same lockup.